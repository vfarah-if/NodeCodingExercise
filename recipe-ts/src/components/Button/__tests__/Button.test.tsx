import React from 'react'
import { render, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import {
  PrimaryButton,
  SecondaryButton,
  LargeButton,
  SmallButton,
  LabellessButton,
} from '../Button.stories'

describe('Button', () => {
  test('should create a primary button', () => {
    const { container } = render(
      <PrimaryButton
        primary={PrimaryButton.args?.primary}
        label={PrimaryButton.args?.label || ''}
      />
    )

    expect(screen.getByRole('button')).toHaveTextContent('Button')
    expect(container.querySelector('button')).toMatchSnapshot()
  })

  test('should create a default medium secondry button', () => {
    const { container } = render(
      <SecondaryButton label={SecondaryButton.args?.label || ''} />
    )

    expect(container.querySelector('button')).toMatchSnapshot()
  })

  test('should create a large secondry button', () => {
    const { container } = render(
      <LargeButton
        size={LargeButton.args?.size}
        label={LargeButton.args?.label || ''}
      />
    )

    expect(container.querySelector('button')).toMatchSnapshot()
  })

  test('should create a small secondry button', () => {
    const { container } = render(
      <SmallButton
        size={SmallButton.args?.size}
        label={SmallButton.args?.label || ''}
      />
    )

    expect(container.querySelector('button')).toMatchSnapshot()
  })

  test('should create a yellow labeless button', () => {
    const { container } = render(
      <LabellessButton
        label={LabellessButton.args?.label || ''}
        style={{ backgroundColor: 'yellow' }}
      />
    )

    expect(container.querySelector('button')).toMatchSnapshot()
  })
})
