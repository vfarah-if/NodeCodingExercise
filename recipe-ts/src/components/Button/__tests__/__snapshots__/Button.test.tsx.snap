// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should create a default medium secondry button 1`] = `
<button
  class="sc-bczRLJ jGJCSX"
  label="Button"
>
  Button
</button>
`;

exports[`Button should create a large secondry button 1`] = `
<button
  class="sc-bczRLJ cGAcOK"
  label="Button"
>
  Button
</button>
`;

exports[`Button should create a primary button 1`] = `
<button
  class="sc-bczRLJ hKYNbp"
  label="Button"
>
  Button
</button>
`;

exports[`Button should create a small secondry button 1`] = `
<button
  class="sc-bczRLJ jcxdVN"
  label="Button"
>
  Button
</button>
`;

exports[`Button should create a yellow labeless button 1`] = `
<button
  class="sc-bczRLJ jGJCSX"
  label=""
  style="background-color: yellow;"
/>
`;
